sdcm/utils/url_validate.py

```diff
@@ -0,0 +1,50 @@
+import re
+
+import requests
+
+SUPPORTED_WEBSITES = ['s3.amazonaws.com', 'scratch.scylladb.com']
+URL_REGEX_AND_FORMAT_LIST = [
+    (re.compile(r'baseurl=(?P<url>http.*?)\$basearch/'), '{url}' + 'x86_64/repodata/repomd.xml'),
+    (re.compile(r'deb\s+\[arch=(?P<arch>.*?)\]\s(?P<url>http.*?)\s(?P<version_code_name>.*?)\s(?P<component>.*?)$'),
+     "{url}/dists/{version_code_name}/{component}/binary-{arch}/Packages"),
+    (re.compile(r'deb\s+(?P<url>http.*?)\s(?P<version_code_name>.*?)\s(?P<component>.*?)$'),
+     "{url}/dists/{version_code_name}/{component}/binary-amd64/Packages.gz"),
+]
+
+
+def verify_url_is_valid(url):
+    """
+    Check the URL of `scylla.list` is valid and verify it's a file.
+    After that check, the file contains a valid scheme and package URL.
+    """
+    supported_content_types, package_urls = ['application/octet-stream', 'binary/octet-stream'], []
```

Comment:
this a very http server specific, I don't think we should care much about those headers...

Reply:
To verify that the URL can be downloaded, it's must contain one of these headers.
Yes, it's very specific but we want to verify each step.
In the future, if we will get failures I'll change this line to less **strict**.